zanata-war/src/main/java/org/zanata/security/ZanataIdentity.java

```diff
@@ -260,10 +616,66 @@ public String login(AuthenticationType authType) {
     public String getAccountUsername() {
         HAccount authenticatedAccount =
                 ServiceLocator.instance().getInstance(
-                        JpaIdentityStore.AUTHENTICATED_USER, HAccount.class);
+                        ZanataJpaIdentityStore.AUTHENTICATED_USER, HAccount.class);
         if (authenticatedAccount != null) {
             return authenticatedAccount.getUsername();
         }
         return null;
     }
+
+    public Principal getPrincipal() {
+        return principal;
+    }
+
+    public void checkRestriction(String expression) {
+        log.warn("permission check as EL is not supported yet");
+        throw new UnsupportedOperationException("identity.checkRestriction(exp) is not supported");
```

Comment:
You are right we don't use @CheckPermission anywhere. There is no @Restrict which uses EL in our code. I think at one stage we have migrated all those into explicit call to identity.checkPermission. 
CheckPermission.value can be used as permission action name in the future. See org/zanata/security/CheckPermissionDecisionVoter.java:59 
I've removed the method ZanataIdentity.checkRestriction(expression). But in ZanatatSecurityInterceptor, it will throw UnsupportedOperationException if someone tries to use the value field. Is this alright? 

----- Original Message -----

> From: "Sean Flanigan" <notifications@github.com>
> To: "zanata/zanata-server" <zanata-server@noreply.github.com>
> Cc: "Patrick Huang" <pahuang@redhat.com>
> Sent: Thursday, July 30, 2015 5:42:59 PM
> Subject: Re: [zanata-server] Replace Seam Security (#908)

> In zanata-war/src/main/java/org/zanata/security/ZanataIdentity.java :
> >          if (authenticatedAccount != null) {
> >              return authenticatedAccount.getUsername();
> >          }
> >          return null;
> >      }
> > +
> > +    public Principal getPrincipal() {
> > +        return principal;
> > +    }
> > +
> > +    public void checkRestriction(String expression) {
> > +        log.warn("permission check as EL is not supported yet");
> > +        throw new
> > UnsupportedOperationException("identity.checkRestriction(exp) is not
> > supported");

> * As far as I know, we don't plan to support EL for security checks at all,
> which means we shouldn't have any annotations which accept EL. EL seems to
> be the purpose of value in @CheckPermission , but we hardly seem to be using
> @CheckPermission anywhere. Is there some migration that we are missing?

> —
> Reply to this email directly or view it on GitHub .